invoice/src/main/java/org/killbill/billing/invoice/InvoiceDispatcher.java

```diff
@@ -700,4 +705,41 @@ public BillingActionPolicy getBillingActionPolicy() {
             return null;
         }
     }
+
+    public void processParentInvoiceForInvoiceGeneration(final ImmutableAccountData account, final UUID invoiceId, final InternalCallContext context) throws InvoiceApiException {
+
+        final InvoiceModelDao invoiceModelDao = invoiceDao.getById(invoiceId, context);
+        final Invoice invoice = new DefaultInvoice(invoiceModelDao);
+
+        BigDecimal invoiceAmount = InvoiceCalculatorUtils.computeInvoiceBalance(invoice.getCurrency(), invoice.getInvoiceItems(), invoice.getPayments());
+        InvoiceModelDao parentInvoice = invoiceDao.getParentDraftInvoice(account.getParentAccountId(), context);
+
+        final Long parentAccountRecordId = internalCallContextFactory.getRecordIdFromObject(account.getParentAccountId(), ObjectType.ACCOUNT, buildTenantContext(context));
+        final InternalCallContext parentContext = new InternalCallContext(context, parentAccountRecordId);
```

Comment:
You need to use `internalCallContextFactory` instead (add the missing API if needed), internal contexts should never be created directly.